examples/image_to_image_controlnet.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+
+from diffusers import (
+    StableDiffusionControlNetImg2ImgPipeline,
+    ControlNetModel,
+    UniPCMultistepScheduler,
+)
+
+from diffusers.utils import load_image
+import numpy as np
+import torch
+
+import cv2
+from PIL import Image
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--base", type=str, default="stabilityai/sd-turbo")
+parser.add_argument("--controlnet", type=str, default="thibaud/controlnet-sd21-canny-diffusers")
+parser.add_argument(
+    "--input_image",
+    type=str,
+    default="https://hf.co/datasets/huggingface/documentation-images/resolve/main/diffusers/input_image_vermeer.png",
+)
+parser.add_argument(
+    "--prompt", type=str, default="chinese painting style women",
+)
+parser.add_argument("--height", type=int, default=512)
+parser.add_argument("--width", type=int, default=512)
+parser.add_argument("--n_steps", type=int, default=7)
+parser.add_argument(
+    "--saved_image", type=str, required=False, default="i2i_controlnet-out.png"
+)
+parser.add_argument("--seed", type=int, default=1)
+parser.add_argument("--warmup", type=int, default=1)
+parser.add_argument("--run", type=int, default=3)
+parser.add_argument(
+    "--compile_unet", type=(lambda x: str(x).lower() in ["true", "1", "yes"]), default=True
+)
+parser.add_argument(
+    "--compile_vae", type=(lambda x: str(x).lower() in ["true", "1", "yes"]), default=True
+)
+parser.add_argument(
+    "--compile_ctrlnet",
+    type=(lambda x: str(x).lower() in ["true", "1", "yes"]),
+    default=True,
+)
+args = parser.parse_args()
+
+# load an image
+image = load_image(args.input_image)
+image = np.array(image)
+
+# get canny image
+image = cv2.Canny(image, 100, 200)
+image = image[:, :, None]
+image = np.concatenate([image, image, image], axis=2)
+canny_image = Image.fromarray(image)
+
+# load control net and stable diffusion
+# reference: https://huggingface.co/docs/diffusers/main/en/api/pipelines/controlnet
+controlnet = ControlNetModel.from_pretrained(args.controlnet, torch_dtype=torch.float16)
+pipe = StableDiffusionControlNetImg2ImgPipeline.from_pretrained(
+    args.base, controlnet=controlnet, torch_dtype=torch.float16
+)
+
+# speed up diffusion process with faster scheduler
+pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
+pipe.to("cuda")
+
+if args.compile_unet:
+    from onediff.infer_compiler import oneflow_compile
+    pipe.unet = oneflow_compile(pipe.unet)
+
+if args.compile_vae:
+    from onediff.infer_compiler import oneflow_compile
+    #pipe.vae = oneflow_compile(pipe.vae)
```

Comment:
Related work takes my time: https://github.com/Oneflow-Inc/onediff/pull/394

But it's not trivial to compile the encode and decode in one graph.